From 4260e83b510423152034c8c77bd99a32afe5e490 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Mon, 05 Apr 2021 19:31:54 +0300
Subject: [PATCH] [SW-22405] gaudi: reset: set compute engines to ref clk

after resetting device or at first boot set
compute engines (TPC and MME) to ref clock (50 MHz).

Change-Id: Id241883b99753e0e53a39df71bde760f244b4d0f
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index aae91ed..54de937 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -178,7 +178,7 @@
 	udelay(HL_PLL_MAX_WAIT_TIME);
 }
 
-static void hl_pll_clk_switch_to_ref(struct hl_pll_cfg *cfg)
+void hl_pll_clk_switch_to_ref(struct hl_pll_cfg *cfg)
 {
 	int sel;
 	u32 reg;
diff --git a/common/pll.h b/common/pll.h
index b58987d..9ff8560 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -52,6 +52,7 @@
 
 void hl_dev_pll_init(void);
 void hl_pll_init(struct hl_pll_cfg *hl_pll_cfg, int size);
+void hl_pll_clk_switch_to_ref(struct hl_pll_cfg *cfg);
 void udelay_init(void);
 u64 hl_pll_timestamp_freq_get(void);
 u32 hl_pll_timestamp_base_get(void);
diff --git a/gaudi/pll.c b/gaudi/pll.c
index 6230cfc..7522f73 100644
--- a/gaudi/pll.c
+++ b/gaudi/pll.c
@@ -117,5 +117,18 @@
 
 void hl_dev_pll_init(void)
 {
+	int i;
+
+	/* PLLs list to be set back to ref clock */
+	struct hl_pll_cfg gaudi_pll_ref_cfg[] =  {
+		{ .base = mmPSOC_MME_PLL_BASE & ~HL_CFG_BASE },
+		{ .base = mmPSOC_TPC_PLL_BASE & ~HL_CFG_BASE },
+		{ .base = mmNIC2_MME_PLL_BASE & ~HL_CFG_BASE },
+		{ .base = mmNIC2_TPC_PLL_BASE & ~HL_CFG_BASE },
+	};
+
 	hl_pll_init(gaudi_pll_cfg, ARRAY_SIZE(gaudi_pll_cfg));
+
+	for (i = 0; i < ARRAY_SIZE(gaudi_pll_ref_cfg); i++)
+		hl_pll_clk_switch_to_ref(&gaudi_pll_ref_cfg[i]);
 }
